ruler/ruler.go

```diff
@@ -51,6 +48,64 @@ type Worker interface {
 	Stop()
 }
 
+type worker struct {
+	delay         time.Duration
+	userID        string
+	configsAPIURL *url.URL
+	opts          *rules.ManagerOptions
+
+	done       chan struct{}
+	terminated chan struct{}
+}
+
+func (w *worker) Run() {
+	defer close(w.terminated)
+	var rs []rules.Rule
+	var group *rules.Group
+	tick := time.NewTicker(w.delay)
+	defer tick.Stop()
+	for {
+		var err error
+		select {
+		case <-w.done:
+			return
+		default:
+			select {
```

Comment:
TBH, it's cargo-culted from similar code in Prometheus. Happy to remove it since it doesn't have tests demonstrating the need for its existence.